src/qibo/core/gates.py

```diff
@@ -558,6 +558,15 @@ def __init__(self, q0, q1):
     def _construct_unitary(self):
         return K.matrices.SWAP
 
+class FSWAP(BackendGate, abstract_gates.FSWAP):
```

Comment:
If you change this to 
```suggestion
class FSWAP(MatrixGate, abstract_gates.FSWAP):
```
it should work with qibojit and qibotf too.

Also please leave two empty lines between this and the previous class to match the rest of the code.

Reply:
I can change this, have no problem with that. However, I do not quite understand why, if the code is like the standard SWAP gate, we cannot somehow use the apply_swap kernel, instead of the generic two qubit gates. I think this would increase performance, but the final word is yours.

Reply:
In order to use the `apply_swap` kernel you need to add an `"fswap": self.apply_swap` entry in the [the dictionary in `AbstractCustomOperators`](https://github.com/qiboteam/qibo/blob/86fa8b76996762aa25fc995b6a2894facc8e0fcf/src/qibo/backends/abstract.py#L624). If you do this, you will stop getting the error you get now with qibojit, but it won't apply the gate you want, it will apply the standard SWAP instead (without the -1 phase). This is because the `apply_swap` kernel does the swap directly, without using the matrix (see [here](https://github.com/qiboteam/qibojit/blob/cbe5f27ce54f7460a6029398ef6d0738b3aa73c5/src/qibojit/custom_operators/gates.py#L199) the kernel code where the `gate` argument is ignored).

There's the option to add a new FSWAP kernel or modify the current SWAP kernel to accomodate the phase change but this needs to be implemented in qibojit first. Another solution is the fSim kernel I wrote above, or just the two-qubit matrix kernel. Let's see what others believe is best.

Reply:
For the moment I have applied the standard two qubit gates and it works with qibojit. We can change it later